Add documentation for draft {P,M}Rs (#14527)
sourcegraph/sourcegraph: doc/user/campaigns/campaign_spec_yaml_reference.md

## [`changesetTemplate.published`](#changesettemplate-published)

Whether to publish the changeset. This may be a boolean value (ie `true` or `false`), `'draft'`, or [an array to only publish some changesets within the campaign](#publishing-only-specific-changesets).

An unpublished changeset can be previewed on Sourcegraph by any person who can view the campaign, but its commit, branch, and pull request aren't created on the code host.

> A draft changeset results in a commit, branch, and pull request / merge request being created on the code host **in draft mode**. This means:

Reviewer: ```suggestion
When `published` is set to `draft` a commit, branch, and pull request / merge request are being created on the code host **in draft mode**. This means:
```

> - On GitHub the pull request will be in draft mode.

Reviewer: ```suggestion
- On GitHub the changeset will be a [draft pull request](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).
```

> - On GitLab the merge request title will be prefixed with `'[Draft] '`, making it unmergeable until it's taken out of draft mode.

Reviewer: ```suggestion
- On GitLab the changeset will be a merge request whose title is be prefixed with `'[Draft] '` to [flag it as a draft merge request](https://docs.gitlab.com/ee/user/project/merge_requests/work_in_progress_merge_requests.html#adding-the-draft-flag-to-a-merge-request).
```

(I don't think you have to mention that it's unmergeable)

> - On BitBucket Server the pull request title will be prefixed with `'WIP: '`.

Reviewer: Is that natively supported by BitBucket Server? If not, I don't think we should do this. Users can already prefix the changeset titles with `WIP: ` if they want to by changing the `changesetTemplate`. And if this doesn't have an effect, it might be more confusing than helpful.

```suggestion
- On BitBucket Server draft pull requests are not supported and changesets published as `draft` won't be created.
```

Author: I found that it will potentially be more confusing that on bitbucket server the action will just be "publish", rather than anything different, when specifying draft mode 🤔 It is not yet supported. It is just a common convention I feel.

Reviewer: > I found that it will potentially be more confusing that on bitbucket server the action will just be "publish", rather than anything different,

Agree. I don't think `draft` on BitBucket Server should mean publish. That's, IMHO, the worst case. Instead "draft" should be a no-op on code hosts that don't support drafts natively. We want to have the least amount of surprising behaviour, especially when it comes to potentially destructive actions (such as force pushing branches). And if I want to be careful and want to roll out drafts first, to test the waters, I'd be surprised if I get a full PR, including CI & notifications to team mates, if one of the 20 repositories is on BitBucket and it doesn't support drafts.

> **Note:** An existing changeset will not be converted to a draft changeset when it already exists. Hence, changesets can only go from unpublished to draft, but not from published to draft. That also allows you to take it out of draft mode on your code host, without risking Sourcegraph to revert to draft mode.

Reviewer: ```suggestion
> NOTE: Changesets that have already been published on a code host as a non-draft (`published: true`) cannot be converted into drafts. Changesets can only go from unpublished to draft to published, but not from published to draft. That also allows you to take it out of draft mode on your code host, without risking Sourcegraph to revert to draft mode.
```

> 

Reviewer: I can't comment on the line (_shakes fist at sky_), but please add one or two examples to the `### Examples` section below.
